testutils/kubernetes.go

```diff
@@ -104,6 +105,53 @@ func KubectlOutAsync(args ...string) (*bytes.Buffer, chan struct{}, error) {
 	return buf, done, err
 }
 
+func KubectlOutPipe(r io.Reader, args ...string) (<-chan *bytes.Buffer, chan struct{}, error) {
```

Comment:
because of the name of this function, i expected to see the use of os pipes somewhere. it's not clear to me what this has to do with pipes. consider renaming the function?

i also see a lot of copy pasta in this file, maybe its time to refactor some of these Kubectl funcs?